solution/index.js

```diff
@@ -0,0 +1,343 @@
+'use strict'
+let taskObj, queryObj;
+const spaceAtEndRegex = /[\s]*$/g;
+let mockList = document.createElement("li");
+let rememberElement;
+
+const addEventListenerToButtons = () => {
+
+    const deleteButtons = document.querySelectorAll('.deleteButton');
+    deleteButtons.forEach(element => {
+        element.addEventListener("click", function () {
+            deleteTask(element.parentElement.parentElement.getElementsByClassName("task")[0].innerText);
+        })
+    });
+
+    const checkButtons = document.querySelectorAll('.checkButton');
+    checkButtons.forEach(element => {
+        element.addEventListener("click", function () {
+            ReplaceTaskType("done", element.parentElement.parentElement.getElementsByClassName("task")[0].innerText);
+        })
+    });
+
+    const sortButtons = document.querySelectorAll('.sortButton');
+    sortButtons.forEach(element => {
+        element.addEventListener("click", function () {
+            element.parentElement.setAttribute("draggable", "true");
+        })
+    });
+}
+
+
+const addEventListenerToTasks = () => {
+    let oldinnerTaskText;
+
+    const taskList = document.querySelectorAll('.task');
+    taskList.forEach(element => {
```

Comment:
There is no justification and it is an inefficient way to add listening events with the help of forEach loop. You should have added listening when creating the task

Reply:
but what about when i reload the page i need to add listener to all of the loaded task